modules/javafx.graphics/src/main/java/com/sun/javafx/iio/common/ImageTools.java

```diff
@@ -146,6 +146,24 @@ public static String getScaledImageName(String path, int scaleFactor) {
         return result.toString();
     }
 
+    public static boolean hasScaledName(String path) {
+        int slash = path.lastIndexOf('/');
+        String name = (slash < 0) ? path : path.substring(slash + 1);
+        int at = name.lastIndexOf('@');
+        if (at < 0) return false;
+
+        int dot = name.lastIndexOf('.');
+        String scale = (dot < 0) ? name.substring(at + 1) : name.substring(at + 1, dot);
+        if (!scale.endsWith("x")) return false;
+
+        try {
+            Integer.valueOf(scale.substring(0, scale.length() - 1));
+        } catch (NumberFormatException ignored) {
+            return false;
+        }
+        return true;
+    }
+
```

Comment:
1. Is looking for a slash going to be compatible on all platforms?  Where is the path string coming from?
2. Catching `NumberFormatException` to "check" if something is a number is bad form
3. It will allow `@0x` and `@-1x` etc...
4. Consider using a regular expression, it is much more concise and intended for this kind of matching

Here's a regular expression for this:

    Pattern SCALED_PATTERN = Pattern.compile(".*@[1-9][0-9]?x(\\.[^\\.]+)?");
    
The above will match any path that ends with `@` followed by a number from 1 to 99, followed by an `x`, optionally followed by an extension that does not contain a dot.

Reply:
If you want you can even return the scale with a slightly altered pattern:

     Pattern SCALED_PATTERN = Pattern.compile(".*@([1-9][0-9]?)x(?:\\.[^\\.]+)?");
     
Then do:

     Matcher matcher = SCALED_PATERN.matcher(path);
     
     if (matcher.matches()) {
          return Integer.parseInt(matcher.group(1));  // can't throw NumbefFormatException, number is validated by pattern
     }
     
     return 0;  // there was no scale

Reply:
>Is looking for a slash going to be compatible on all platforms? Where is the path string coming from?

My assumption was that other parts of ImageTools use this assumption so here it would be okay as well. `getScaledImageName` right above this method assumes so (although there it might not matter as much).

Regex might be a better fit for this, I agree.